WalletWasabi/Services/MemPoolService.cs

```diff
@@ -222,12 +224,18 @@ public async Task<bool> TryPerformMempoolCleanupAsync(NodesGroup nodes, Cancella
 					}
 				}
 
-				uint256[] toRemove = TransactionHashes.Except(allTxs).ToArray();
-				foreach (uint256 tx in toRemove)
+				int removedTxCount = 0;
+				foreach (uint256 tx in TransactionHashes)
 				{
-					TransactionHashes.TryRemove(tx);
+					if(!allTxs.Contains(tx))
+					{
+						if(TransactionHashes.TryRemove(tx))
```

Comment:
Is it safe to remove an item during iterate? At normal collections this can cause `IOE (Collection modified)`. Maybe the same exists in concurrent collections. I would use a temporary list as we did before.

Reply:
Yeah, I think the concurrent collections won't throw, but it's not a good idea to do it regardless.

Reply:
The `ConcurrentHashSet::GetEnumerator` says:
> The enumerator returned from the collection is safe to use concurrently with reads and writes to the collection.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/1533/commits/a795276b73a0997d01c6e8cd9a31b69616712ddb